sysutils/pfSense-pkg-RRD_Summary/files/usr/local/www/status_rrd_summary.php

```diff
@@ -20,40 +20,87 @@
-	return explode('|', trim($traffic[0]));
+	exec("/usr/local/bin/rrdtool fetch {$rrd} AVERAGE -r {$resolution} -s {$start} -e {$end}", $traffic);
+	$divisor = 1024 ** $units;
+	$t_keys = preg_split("/\s+/", $traffic[0]);	// grab keys
+	//print "time=$t_first st=$start end=$end div=$divisor res=$resolution";
+	$traffic = preg_grep("/^[0-9]+:/", $traffic);	// select data rows
+	$traffic = preg_grep("/nan/", $traffic, PREG_GREP_INVERT);	// filter nan rows
+	$data = array( "first" => time() );
+	foreach ( $traffic as $t ) {
+		$t = preg_split("/[\s:]+/", $t);
+		if (count($t) != count($t_keys)) continue;	// error: field mismatch
+		$data["first"] = min($t[0], $data["first"]);
+		$data["last"] = max($t[0], $data["last"]);
+		foreach ($t_keys as $i => $k) { $data[$k] += ($t[$i] / $divisor) * $resolution; }
+	}
+	// Adjust for resolution
+	foreach ($t_keys as $k) {
+		if (preg_match("/^outblock/", $k)) continue;
+		$data["total_in"] += preg_match("/^in/", $k) ? $data[$k] : 0;
+		$data["total_out"] += preg_match("/^out/", $k) ? $data[$k] : 0;
+	}
+	return $data;
 }
 
-function print_rrd_summary_table($data) { ?>
-		<div class="table-responsive">
-			<table class="table table-striped table-hover table-condensed" id="rrdsummary">
-				<thead>
-					<tr><th>Direction</th><th>Bandwidth</th></tr>
-				</thead>
-				<tbody>
-					<tr><td>In</td><td align="right"><?=$data[0]; ?> MBytes</td></tr>
-					<tr><td>Out</td><td align="right"><?=$data[1]; ?> MBytes</td></tr>
-					<tr><td>Total</td><td align="right"><?=$data[0] + $data[1]; ?> MBytes</td></tr>
-				</tbody>
-			</table>
-		</div>
-<?php
+function print_rrd_summary($rrd, $units, $startyear, $startday) {
+	$data = fetch_rrd_summary($rrd, "-10y", "now");
+	$first = $data["first"];
+	$last = $data["last"];
+	global $unitlist; $u = array_flip($unitlist);
+	foreach (range(date("Y", $last), date("Y", $first), 1) as $year) {
+		if ($startyear > 0 && $startyear != $year) continue;
+		foreach (range(12, 1, -1) as $month) {
+			$start = strtotime(date("{$month}/{$startday}/{$year}"));
+			$end = strtotime("-1 second", strtotime("+1 month", $start));
+			if ($start > $last || $end < $first) continue;
+			if ($start < $first) $start = $first;
+			if ($end > $last) $end = $last;
+			$data = fetch_rrd_summary($rrd, "epoch+{$start}s", "epoch+{$end}s", $units, 24*60*60);
+			?>
+									<tr>
+										<td><?=date("Y-m-d", $start); ?> to <?=date("m-d-Y", $end); ?></td>
+										<td><?=sprintf("%0.2f %s", $data["total_in"], $u[$units]); ?></td>
+										<td><?=sprintf("%0.2f %s", $data["total_out"], $u[$units]); ?></td>
+										<td><?=sprintf("%0.2f %s", $data["total_in"] + $data["total_out"], $u[$units]); ?></td>
+									</tr>
```

Comment:
Indenting looks excessive

Reply:
I agree, but felt it was better to preserve the original indenting than fix it here.  Indenting is fixable, let me know if you'd like me to fix it.

Reply:
Yes please. It is good to take the opportunity to fix things like this and get the codebase as consistent as possible :) Nice changes BTW.

Reply:
Thanks!  Will make the changes.  Would you like me to squash the commit?